src/ray/rpc/worker/core_worker_client.cc

```diff
@@ -62,6 +62,12 @@ void CoreWorkerClient::PushActorTask(std::unique_ptr<PushTaskRequest> request,
 
   {
     absl::MutexLock lock(&mutex_);
+    if (max_finished_seq_no_ == absl::nullopt) {
+      max_finished_seq_no_ = request->sequence_number() - 1;
+    }
+    // The RPC client assumes that the first request put into the send queue will be
```

Comment:
1. [SendPendingTasks](https://github.com/ray-project/ray/blob/ad4e989ebffd7a0d4068dd891be79f032082164c/src/ray/core_worker/transport/actor_task_submitter.cc#L557) calls PushActorTask for tasks in order from the smallest to the largest sequence number.

2. With this PR, the sequence number is strictly increasing, so the sequence numbers of resubmitted tasks will be updated before resubmission.


https://github.com/ray-project/ray/blob/ad4e989ebffd7a0d4068dd891be79f032082164c/src/ray/core_worker/core_worker.cc#L1389